File: 01_basic_structure/sample.c
// C언어의 주요 특징
// 1. 절차지향적 언어 (Procedural Programming Language)
// 2. 하드웨어를 직접 제어할 수 있는 시스템 프로그래밍에 적합
// 3. 이식성이 높고 실행 속도가 빠름
// 4. 모듈화 프로그래밍 지원

// C언어 프로그램의 기본 구조
// 1. 전처리기 지시문 (#include, #define 등)
// 2. 전역 변수 선언 (필요한 경우)
// 3. 함수 정의 (main 함수 포함)
// 4. 프로그램 종료 (return 문)

#include <stdio.h>  // 표준 입출력 함수를 위한 헤더
#include <stdlib.h> // 메모리 할당, 난수 생성 등을 위한 헤더

/*
    main 함수
    - 모든 C 프로그램의 필수 진입점(entry point)
    - int main(void): 매개변수 없음, 정수 반환
    - 프로그램이 정상 종료되면 0을 반환
*/
int main(void)
{
    printf("Hello, C\n");  // 화면에 문자열 출력

    return 0;  // 프로그램 정상 종료
}

File: 01_basic_structure/Sample.java
// Java 언어의 주요 특징
// 1. 객체지향적 언어 (Object-Oriented Programming Language)
// 2. 플랫폼 독립적 (Write Once, Run Anywhere)
// 3. 자동 메모리 관리 (Garbage Collection)
// 4. 풍부한 표준 라이브러리 제공 (Java API)
// 5. 멀티쓰레드 프로그래밍 기본 지원
// 6. 동적 로딩을 통한 유연한 프로그램 실행

// Java 프로그램의 기본 구조
// 1. 패키지 선언 (package statement)
// 2. import문 선언 (필요한 클래스 import)
// 3. 클래스 선언 (public class)
// 4. main() 메소드 선언 (프로그램 진입점)

/*
    Sample 클래스
    - Java 프로그램의 기본 구조를 보여주는 예제
    - public class는 파일명과 동일해야 함
    - 모든 Java 프로그램은 최소 하나의 클래스가 필요
*/
public class Sample {
    /*
        main 메소드
        - Java 프로그램의 필수 진입점(entry point)
        - public static void main(String[] args): 표준 시그니처
        - args: 명령행 매개변수를 저장하는 문자열 배열
    */
    public static void main(String[] args) {
        System.out.println("Hello, Java");  // 표준 출력 스트림에 문자열 출력
    }
}
